File: .1/main.py
from typing import List

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        result = []
        a = 0
        n = len(nums)
        for i in range(n):
            for j in range(i + 1, n):
                a = nums[i] + nums[j]
                if a == target:
                    return [i, j]
        

File: .1/main.go
package main

// import "fmt"

func twoSum(nums []int, target int) []int {
	seen := make(map[int]int)

	for i, num := range nums {
		diff := target - num
		// これについて(j, ok = seen[diff])という書き方ができるん？
		if j, ok := seen[diff]; ok {
			return []int{i, j}
		}
		// これってなんだっけ？
		seen[num] = i
	}
	// 何もなかったらnilを返すのってないとマズイの？
	return nil
}